src/ape/api/projects.py

```diff
@@ -95,11 +95,29 @@ def cached_manifest(self) -> Optional[PackageManifest]:
         if it exists and is valid.
         """
         if self._cached_manifest is None:
-            manifest = _load_manifest_from_file(self.manifest_cachefile)
-            if manifest is not None:
-                manifest.contract_types = self.contracts
-            self._cached_manifest = manifest
-        return self._cached_manifest
+            self._cached_manifest = _load_manifest_from_file(self.manifest_cachefile)
+            if self._cached_manifest is None:
+                return None
+
+        manifest = self._cached_manifest
+        if manifest.contract_types and not self.contracts:
+            # Extract contract types from cached manifest.
+            # This helps migrate to >= 0.6.3.
```

Comment:
Attitude note to remove in 0.7.0

Reply:
OMG, "Add a TODO note" = "Attitude note", always now.

Reply:
lmfao whoops